src/System.IO.FileSystem/tests/FileInfo/GetSetTimes.cs

```diff
@@ -66,6 +67,57 @@ public override IEnumerable<TimeFunction> TimeFunctions(bool requiresRoundtrippi
                 DateTimeKind.Utc);
         }
 
+        [Fact]
+        public void CopyToMillisecondPresent()
+        {
+            FileInfo input = new FileInfo(Path.Combine(TestDirectory, GetTestFileName()));
+            input.Create().Dispose();
+
+            string driveFormat = new DriveInfo(input.DirectoryName).DriveFormat;
+            if (!driveFormat.Equals(HFS, StringComparison.InvariantCultureIgnoreCase))
+            {
+                for (int i = 0; i < 5; i++)
+                {
+                    if (input.LastWriteTime.Millisecond != 0)
+                        break;
+
+                    // This case should only happen 1/1000 times, unless the OS/Filesystem does
+                    // not support millisecond granularity.
+
+                    // If it's 1/1000, or low granularity, this may help:
+                    Thread.Sleep(1234);
+                    input = new FileInfo(Path.Combine(TestDirectory, GetTestFileName()));
+                    input.Create().Dispose();
```

Comment:
Why not just call input.Create().Dispose() again on the original input FileInfo?